### ISSUE TITLE
Unable to set EdgeDeflector as the default for microsoft-edge: links (not shown as an option in Default apps)

### ISSUE THREAD

**Issue by gnpaone** [da2x/EdgeDeflector#141]
I recently updated to build 22494 (windows dev build) and edgedeflector not being set as default in settings and it's not opening preferred browser anymore and this app is not even in the list of "MICROSOFT-EDGE" link opener. I reinstalled and followed all the [troubleshooting](https://github.com/da2x/EdgeDeflector/wiki/Troubleshooting) steps given by you, still facing the problem. Please help!
![Screenshot (4770)](https://user-images.githubusercontent.com/78990165/140705640-45d4bd9d-8b8f-4484-9300-fad1c51ef5ab.png)
![Screenshot (4771)](https://user-images.githubusercontent.com/78990165/140705648-47609329-19b9-46d0-b14b-83a2a55fc4e4.png)

**Comment by Jonstep101010** [comment]
yup, same case here.

**Comment by AlexisLust** [comment]
same case here

**Comment by da2x** [comment]
I’m on it.

**Comment by da2x** [comment]
Microsoft has [made changes in Windows 11 build 22949](https://www.ctrl.blog/entry/microsoft-edge-protocol-competition.html) to block EdgeDeflector and other apps from handling `microsoft-edge:` links. I've had some backup plans for such an eventuality, but Microsoft has been more aggressive than I predicted. I don't have any non-destructive workarounds at this time.

**Comment by gnpaone** [comment]
> Microsoft has [made changes in Windows 11 build 22949](https://www.ctrl.blog/entry/microsoft-edge-protocol-competition.html) to block EdgeDeflector and other apps from handling `microsoft-edge:` links. I've had some backup plans for such an eventuality, but Microsoft has been more aggressive than I predicted. I don't have any non-destructive workarounds at this time.

Ohh, It's bad that Microsoft is sabotaging user preferences, I use Edgedeflector from so long and hoping that you will come with a solution soon ☺️

**Comment by rcmaehl** [comment]
Hi @da2x 

Check out my repo https://github.com/rcmaehl/MSEdgeRedirect

If fixes the issue you're having, doesn't require changing default apps or protocol handlers, and It works with Cortana as well! I'd love to see this implementation ported to the existing, more popular solution

**Comment by sagudev** [comment]
According to [article](https://www.ctrl.blog/entry/microsoft-edge-protocol-competition.html) you can still select Edge insiders for handlers. So how does the Edge registers itself as MS Edge? (Observing Edge insiders instalation might give us a solution). If we could register other app (egg. EdgeDeflector) as Edge in the system, then system will allow to choose EdgeDeflector as ` microsoft-edge:// ` handler.

**Comment by linux-leo** [comment]
what if we just replace the msedge.exe with a programm that can accept the url and redirect it to another browser? It would probably turn into EdgeRemover. But people that use this don't plan on using edge anyway.

**Comment by anaisbetts** [comment]
You probably don't need to do any of this stuff, use Image File Execution Options's `Debugger` key to set the "Debugger" of msedge.exe to EdgeDeflector. You get the path that msedge.exe was called with and you can instead execute whatever browser you want.

**Comment by SerraraFluttershy** [comment]
Couldn't you also just rename msedge.exe to something else and place a new EXE with its name in its place?

**Comment by gnpaone** [comment]
> Couldn't you also just rename msedge.exe to something else and place a new EXE with its name in its place?

No it's not possible, each program is paired with its own program id and hash and is registered by Windows so Windows checks these details before assigning this app to open something to avoid duplicates/malware/unauthorised access etc. In 22494 Windows locked microsoft-edge link to use edge (and it's other edge builds) via program id and hash so the programs matching these details only can perform the actions, it's read-only so can't change, if tried to change then computer won't open as it fast verifies on every login

**Comment by Wolfhound905** [comment]
You made it onto the tech news XD: https://youtu.be/2EyLR42Ko6I?t=18

**Comment by soredake** [comment]
https://github.com/rcmaehl/MSEdgeRedirect

**Comment by Suncoast-Cyclist** [comment]
I just noticed that Deflector is still working on my Windows 11 Dev preview system after updating to build 22499.1010.  Apparently Microsoft is not resetting the current association back to Edge when you upgrade.  I no longer have the option of changing it to anything but one of the Edge browser versions, but as long as I leave everything alone, Deflector continues to work.  It does give me the option of selecting the version of Deflector that is available for purchase in the Microsoft Store.

**Comment by Smart123s** [comment]
>  It does give me the option of selecting the version of Deflector that is available for purchase in the Microsoft Store.

If that's really the case, maybe an .appx package of ED could work, even if it's side-loaded.

Although I'm pretty sure MS will eventually fix it as well.

**Comment by Iwitrag** [comment]
Microsoft is breaking a law there by forcing users to use their browser. Lawsuit imminent.

**Comment by Spectroxx** [comment]
> 
> 
> Microsoft is breaking a law there by forcing users to use their browser. Lawsuit imminent.

The EU might have some strong opinions about forcing users to use Edge

**Comment by JovietUnion** [comment]
> > Microsoft has [made changes in Windows 11 build 22949](https://www.ctrl.blog/entry/microsoft-edge-protocol-competition.html) to block EdgeDeflector and other apps from handling `microsoft-edge:` links. I've had some backup plans for such an eventuality, but Microsoft has been more aggressive than I predicted. I don't have any non-destructive workarounds at this time.
> 
> Ohh, It's bad that Microsoft is sabotaging user preferences, I use Edgedeflector from so long and hoping that you will come with a solution soon ☺️

Well, at least you can choose not to upgrade to windows 11. Or if you are lucky enough to not meet hardware requirements, just wait til he fixes the issue.

**Comment by CrimsonXMC** [comment]
> > > Microsoft has [made changes in Windows 11 build 22949](https://www.ctrl.blog/entry/microsoft-edge-protocol-competition.html) to block EdgeDeflector and other apps from handling `microsoft-edge:` links. I've had some backup plans for such an eventuality, but Microsoft has been more aggressive than I predicted. I don't have any non-destructive workarounds at this time.
> > 
> > 
> > Ohh, It's bad that Microsoft is sabotaging user preferences, I use Edgedeflector from so long and hoping that you will come with a solution soon ☺️
> 
> Well, at least you can choose not to upgrade to windows 11. Or if you are lucky enough to not meet hardware requirements, just wait til he fixes the issue.

i wouldn't be surprised if ms backports this to windows 10 in the near future

**Comment by Spectroxx** [comment]
> 

Let's hope not. I'm sure there's a way around to this issue..? right...?

**Comment by JovietUnion** [comment]
Not sure, m8.
If this does happen, I'm moving to Linux.

**Comment by trumad** [comment]
Tangenttially related, Edge will now include [bloatware](https://www.xda-developers.com/microsoft-edge-buy-now-pay-later/) which adds a "buy now pay later" drop-down to all payment options on all online storefronts. Are they making more of an effort to lock people in to edge so that they can push them towards predatory loans like this?

**Comment by ZaLiTHkA** [comment]
> i wouldn't be surprised if ms backports this to windows 10 in the near future

looking at #151, I'd say this is already happening.. 😢

**Comment by fariage** [comment]
Traumatized =(
Please make EdgeDeflect work again...

**Comment by meek2100** [comment]
Same, KB5007262 killed Edge Deflector for me.  No way to set it as default again.

OS Build: 22000.348

**Comment by HarshalKudale** [comment]
> what if we just replace the msedge.exe with a programm that can accept the url and redirects it to another browser? It would probably turn into EdgeRemover. But people that use this don't plan on using edge anyway.

I have tried this it works but stops working after restart as windows will try to assign protocol back to msedge.exe the we have msedge.exe with different progid and hash so it breaks and then we cant open edge and windows search.

**Comment by HarshalKudale** [comment]
> You probably don't need to do any of this stuff, use Image File Execution Options's `Debugger` key to set the "Debugger" of msedge.exe to EdgeDeflector. You get the path that msedge.exe was called with and you can instead execute whatever browser you want.

I have tried this and its working.I did it in cpp . My cpp program also converts bing to google. Currently I am working on automating the manual work(registry stuff for IFE debugger) as well as fixing the recursion when bare edge is opened (fixed it). 

So basically 
- it will redirect windows search to default browser with bing to google conversion.
- Can open edge instead of default browser (if required)
- Its fast 
- Need some manual setup (automating it)

**Comment by fariage** [comment]
@HarshalKudale , thanks for the feebback. Given the fact that I´m no coder, can you pleas give us step-by-step instructions to make EdgeDeflector work again?. Thanks a lot in advance, really.

PD:For some extra motivation: you can do that as a birthday present to me!!! ;-)

**Comment by vuon9** [comment]
https://github.com/AveYo/fox/blob/main/ChrEdgeFkOff.cmd here is a way @fariage - play at your own risk.

**Comment by HarshalKudale** [comment]
> @HarshalKudale , thanks for the feebback. Given the fact that I´m no coder, can you please give us step-by-step instructions on how to make EdgeDeflector work again?. Thanks a lot in advance, really.
> 
> PD:For some extra motivation: you can do that as a birthday present to me!!! ;-)

I have ported my app to c# for easy to use and to prevent console window from opening.
I will be uploading code and executable om github.
soon(probably this week). 

Current features. 
- Can redirect windows search to default browser
- Converts bing to google
- Fast
- Edge can be opened normally.(if someone want to use edge along side my software).
- Also works with simple urls like microsoft-edge://https://google.com

Currently working one
- automating registry setup
- automating executable setup.

I need more sample urls with Microsoft-edge url if anyone know url besides windoes search please comment here. I need to implement those to prevent app from breaking with different urls.

**Comment by HarshalKudale** [comment]
> @HarshalKudale , thanks for the feebback. Given the fact that I´m no coder, can you please give us step-by-step instructions on how to make EdgeDeflector work again?. Thanks a lot in advance, really.
> 
> PD:For some extra motivation: you can do that as a birthday present to me!!! ;-)

Its done check my github.

**Comment by JovietUnion** [comment]
Take that, Microsoft!

**Comment by fariage** [comment]
Thanks for the effort @HarshalKudale , but NoMoreEdge 1.0.0.2 is not working for me =(

**Comment by HarshalKudale** [comment]
> Thanks for the effort @HarshalKudale , but NoMoreEdge 1.0.0.2 is not working for me =(

Please get the latest release (v1.0.1.0) from repo. And create issue on github page if anything is not working.

**Comment by sudological** [comment]
Maybe this should be a separate issue, I'm not sure. But the same behavior is caused by Windows 10 update KB5007253. I'm guessing that it works in the same or a similar way, but that is speculation on my part.

**Comment by HarshalKudale** [comment]
> Maybe this should be a separate issue, I'm not sure. But the same behavior is caused by Windows 10 update KB5007253. I'm guessing that it works in the same or a similar way, but that is speculation on my part.

Microsoft is most probably backporting the restrictions to windows 10 . can you provide steps to reproduce the issue?

**Comment by HarshalKudale** [comment]
> @HarshalKudale Steps to reproduce:
> 
> 1. Install the update
> 2. Try to use Windows search.
> 3. Your search will go to Edge. :(
> 
> Steps to fix:
> 
> 1. Go to Windows Update
> 2. Uninstall that particular update.
> 3. Try to use Windows search.
> 4. You can use Edge Deflector again.

Edge deflector will not work on new windows update. You can use other utilities like NoMoreEdge(my app) or Ms edge redirect .

**Comment by J3fftw1** [comment]
Will EdgeDeflector work again with the newest Win update since they reverted the setting to change default app?
As stated in this article from bleeping [article](https://www.bleepingcomputer.com/news/microsoft/microsoft-reverses-windows-11s-annoying-default-browser-setting-changes/)

**Comment by rcmaehl** [comment]
> Will EdgeDeflector work again with the newest Win update since they reverted the setting to change default app? As stated in this article from bleeping [article](https://www.bleepingcomputer.com/news/microsoft/microsoft-reverses-windows-11s-annoying-default-browser-setting-changes/)

As far as I'm aware that's for the default browser. I did not see anything about them reverting URI restrictions.

**Comment by wickedyoda** [comment]
I had the same issue but found online where MS had pushed a patch that disabled edge from being taken over by other browsers forcing people to use it. Since the push, there was so much conflict that MS reports now they have pushed another patch removing the previous. 

So far the app doesn't work as edge still doesn't let you change it, I wonder what the patch coming in the next week or two will allow. Maybe the person above being on the DeV version already has the patch and that's what he is seeing. Since this is my main machine, I won't be visiting the Dev/insider with it. Might check on one of my other systems.

**Comment by wickedyoda** [comment]
> Will EdgeDeflector work again with the newest Win update since they reverted the setting to change default app? As stated in this article from bleeping [article](https://www.bleepingcomputer.com/news/microsoft/microsoft-reverses-windows-11s-annoying-default-browser-setting-changes/)

Haha, I read a lot of the above post and should have read more. Pretty much what I said you had already.

**Comment by HarshalKudale** [comment]
> > Will EdgeDeflector work again with the newest Win update since they reverted the setting to change default app? As stated in this article from bleeping [article](https://www.bleepingcomputer.com/news/microsoft/microsoft-reverses-windows-11s-annoying-default-browser-setting-changes/)
> 
> Haha, I read a lot of the above post and should have read more. Pretty much what I said you had already.

They did not remove restrictions of microsoft-edge protocol . Microsoft only added new setting that lets you choose default browser in one on go.

So even in new patch edge deflector will not work. But on the other hand apps like NomoreEdge and Ms edge redirect will work because they use different methods to grab the protocol url from Edge. These apps do not rely on Microsoft-edge protocol access instead they use IFEO.

**Comment by eturk1** [comment]
Windows 10 Build 19043 broke EdgeDeflector for me. LIke a read above, it's removed from the list of recognized apps for MICROSOFT-EDGE protocol.

Interesting the "Look for an app in the Microsoft Store" gets me to two apps. I wonder if they work, since they're found by some attribute in the store. Maybe put this app in the store like theirs (which are years old)?

**Comment by dtantono** [comment]
@eturk1 there is an issue for that and it is been abruptly closed without a reason.  https://github.com/da2x/EdgeDeflector/issues/151#issuecomment-979687480

**Comment by rcmaehl** [comment]
> Interesting the "Look for an app in the Microsoft Store" gets me to two apps. I wonder if they work, since they're found by some attribute in the store.

They do not

![image](https://user-images.githubusercontent.com/716581/145721572-58765e95-b34e-4d64-8841-c728d1f1e247.png)

**Comment by MagicPlants** [comment]
> Windows 10 Build 19043 broke EdgeDeflector for me. LIke a read above, it's removed from the list of recognized apps for MICROSOFT-EDGE protocol.

Same for me! I'm super disappointed...

**Comment by sero583** [comment]
Since my latest update also on Windows 10 now. MS forces me to use Edge...

**Comment by HarshalKudale** [comment]
Only possible way to redirect Edge is to use IFEO or have process continuously running and redirecting links to default browser.

**Comment by dtantono** [comment]
I remember that the addon that use IFEO is still in Beta. Is it stable?

**Comment by fireton** [comment]
I switched to MSEdgeRedirect.

**Comment by HarshalKudale** [comment]
> I switched to MSEdgeRedirect.

Good choice

**Comment by 2br-2b** [comment]
Is there a way to change registry values to make EdgeDeflector the default choice (even if it can't be done via a gui)?

**Comment by eturk1** [comment]
> I switched to MSEdgeRedirect.



> I switched to MSEdgeRedirect.

Thanks, It works great!
Just use "active" mode because the default mode was broken for me and opened hundreds of browser tabs.
Got immediate response on my bug report though.

**Comment by dtantono** [comment]
@eturk1 it is still in beta and it is still usable for Win10. But, don't expect it too much though.

**Comment by webluke** [comment]
I just tried on Windows 10 (19044.1415), and my Microsoft-Edge is all confused still opens Edge, but I must have set it to Brave at some point, but MS is doing some trickery. 
![image](https://user-images.githubusercontent.com/5162188/147417985-24dc5981-f2f8-4a54-b0a4-fa925441b9a5.png)

**Comment by peterdsharpe** [comment]
Hi all,

I think I have a fix that works in the interim.

By force-uninstalling Edge, I find that on my machine (latest Windows 11 Home 22000.376), links get redirected through Chrome (my default browser). 

To force-uninstall Edge, [follow the steps here](https://www.techradar.com/how-to/how-to-remove-microsoft-edge-from-windows-10). Copy-pasting from the link:

> Removing Microsoft Edge in Command Prompt is a two-step process. First, you need to navigate to the correct folder using this command:
> 
> ```cd %PROGRAMFILES(X86)%\Microsoft\Edge\Application\xx\Installer```
>
> When entering it, substitute ‘xx’ for the current version number of Microsoft Edge installed on your PC. You can find that information in the ‘About’ section of Edge’s settings. 
> 
> Your Command Prompt will now change to show that you’re in the folder you navigated to above. Now, enter this command:
> 
> ```setup --uninstall --force-uninstall --system-level```
> 
> There’s no restart involved, Microsoft Edge will now be removed from your system. You may still see it in the Start Menu, but it won’t open anything and the annoying ‘Restore recommended’ for web browsing in the Settings app will have gone.

**Comment by JovietUnion** [comment]
Antitrust maybe

**Comment by billie-sc** [comment]
> 

done it but wont search for anything put in the start menu. kinda fustrating but whatever :/

**Comment by Hassnjwr6** [comment]
> [#141 (comment)](https://github.com/da2x/EdgeDeflector/issues/141#issuecomment-1003748656)
> 
> done it but wont search for anything put in the start menu. kinda frustrating but whatever :/

I face same as your issue/problem , I am on win10 build 19044.1466
First I nstalled //EdgeDeflector_install.exe, no luck it redirects me to M-Edge, Prior to this problem I just uninsalled Firefox and deleted everything related to, %appdata%, Roaming , Local , every Mozila file, Because my Firefox behaving laggy and broken some times in websites such (youtube)
Seconed I uninstalled M-Edge, then tried to type anything on Search/cortana then nothing happens, it should redirect me to firefox my defult browser, tried to restart my PC several time no luck. 
Also, the window that prompting me to choose Edge-def. did not show at all, even when i change the defult app 
*Settings >> Default Apps >> Web browser (Firefox)
*// // Choose default apps by protocol , kept every M-edge same as what it is, even if I attempt to change it is only prompting M-Edge to me and "Look for an app ...". 

[Solution] 
I went to redownload Edge again https://www.microsoft.com/en-us/edge
tried to type on cortana/search it redirected me back to firefox. 
Also went to *Settings >> Default Apps >> Choose default apps by protocol 
Firefox is the defult on protocol such HTTP, HTTPS which was not. 

In my opinon, I think there is some problem of priority in regestery that leads prevent any change of default app of web browser. 


tl;dr
I uninstalled M-edge, installed EdgeDeflector, Reinstall M-edge , Problem solved. 

Hope this Fixes your problem.

**Comment by NilsMoller** [comment]
> 

sadly, this did not work for me (on the same windows 10 build)

**Comment by timstjiang** [comment]
Having the same issue on Win 10 Pro 19044.1526.
I explored some options such as SetUserFTA. It was able to change the default protocol to chrome, but pages wouldn't open and Windows throws an error.
Is there a way to find the progid of EdgeDeflector? Since that is what SetUserFTA uses to set default apps and protocols.

**Comment by zhufucdev** [comment]
> You probably don't need to do any of this stuff, use Image File Execution Options's `Debugger` key to set the "Debugger" of msedge.exe to EdgeDeflector. You get the path that msedge.exe was called with and you can instead execute whatever browser you want.

So I implemented it as a program [here](https://github.com/zhufucdev/EdgeRemover)

**Comment by Eagle3386** [comment]
@zhufucdev Doesn't work here - steps to reproduce:
1. Downloaded `EdgeRemover.exe` from your repo's releases tab (tagged as `polished`).
2. Put the application next to EdgeDeflector (`C:\Users\[MyUserName]\AppData\Local\Programs\EdgeDeflector`)
3. Opened PowerShell (as admin), "`cd`'ed" into aforementioned directory & ran `.\EdgeRemover.exe`.
4. Got the "success message box":
![image](https://user-images.githubusercontent.com/5111490/158966411-9ba62688-c7c2-4f0d-b854-bf1330bc96dd.png)
5. Confirmed that the registry's value is set correctly:
![image](https://user-images.githubusercontent.com/5111490/158966744-e73d66f5-7b1f-4caa-b33f-dfa086d8e1d8.png)
6. Went into settings, default apps by protocol, only to see 2 selectable Edge entries:
![image](https://user-images.githubusercontent.com/5111490/158967253-499e6122-af00-4d8a-8909-cf049454ee07.png)
7. Even set all Edge-related items back to the entry with Edge's icon.

Yet, the result is always the same: no Edge or any other browser (-tab) is launched - neither from taskbar's news flyout, nor EdgeDeflector's "test link", `microsoft-edge:https://www.ctrl.blog`, nor any other Edge-forced link I could find & try.

Any help would be greatly appreciated...

**Comment by zhufucdev** [comment]
> 

I am currently on Windows 11 and failed to reproduce your issue. Hence I need more information. Is there a `edge-remover.lock` file under the EdgeDeflector folder?
Besides, it is a different project, independent of EdgeDeflector. Hence all problems were supposed to be under its own issues section

**Comment by Eagle3386** [comment]
Yes, it's there & its content is `2022-03-18 10:16:33`.
Also opened issue in your repo: zhufucdev/EdgeRemover#2.

**Comment by Eagle3386** [comment]
For anyone interested, there's a _working_ alternative here on GitHub, available over at [rcmaehl/MSEdgeRedirect](https://github.com/rcmaehl/MSEdgeRedirect).
Using it's `Active Mode`, Edge isn't even launched prior to redirection, i. e. it works just as EdgeDeflector did previously. 🥳

Bonus: it does de-bloating of Bing searches, MSN news, MSN weather, Edge-forced PDF, etc.
